### PR TITLE
Switch to conda-forge PyTorch, install only conda-forge libs (pyg-lib, torch-scatter)

### PR DESCRIPTION
**Pull request by alexbarghi-nv**

### CODE REVIEW

File: ci/test_python.sh [hunk 1]
```diff
@@ -210,23 +210,15 @@ if [[ "${RAPIDS_CUDA_VERSION}" == "11.8.0" ]]; then
     rapids-mamba-retry install \
       --channel "${CPP_CHANNEL}" \
       --channel "${PYTHON_CHANNEL}" \
-      --channel pytorch \
       --channel pyg \
       --channel nvidia \
+      --channel conda-forge \
```
[the remark this answers is not in the document]
Comment (alexbarghi-nv) [reply ci/test_python.sh:215]: yeah, I can do that

**Comment by bdice** [comment]
With this PR, do we still need to skip tests on CUDA 12? I think there should be packages available for CUDA 12 on the new channels. If not, we should identify what is missing.

```
 RAPIDS logger » [04/17/24 20:09:23]
┌─────────────────────────────────────────────────┐
|    skipping cugraph_dgl pytest on CUDA!=11.8    |
└─────────────────────────────────────────────────┘


RAPIDS logger » [04/17/24 20:09:23]
┌─────────────────────────────────────────────────┐
|    skipping cugraph_pyg pytest on CUDA!=11.8    |
└─────────────────────────────────────────────────┘


RAPIDS logger » [04/17/24 20:09:23]
┌─────────────────────────────────────────────────────────┐
|    skipping cugraph-equivariant pytest on CUDA!=11.8    |
└─────────────────────────────────────────────────────────┘
```

**Comment by alexbarghi-nv** [comment]
> With this PR, do we still need to skip tests on CUDA 12? I think there should be packages available for CUDA 12 on the new channels. If not, we should identify what is missing.
> 
> ```
>  RAPIDS logger » [04/17/24 20:09:23]
> ┌─────────────────────────────────────────────────┐
> |    skipping cugraph_dgl pytest on CUDA!=11.8    |
> └─────────────────────────────────────────────────┘
> 
> 
> RAPIDS logger » [04/17/24 20:09:23]
> ┌─────────────────────────────────────────────────┐
> |    skipping cugraph_pyg pytest on CUDA!=11.8    |
> └─────────────────────────────────────────────────┘
> 
> 
> RAPIDS logger » [04/17/24 20:09:23]
> ┌─────────────────────────────────────────────────────────┐
> |    skipping cugraph-equivariant pytest on CUDA!=11.8    |
> └─────────────────────────────────────────────────────────┘
> ```

I just reenabled CUDA 12 testing.  Let's see what happens.

**Comment by alexbarghi-nv** [comment]
CUDA 12 testing still seems to be blocked because our CI is using CUDA 12.2 but there are only CUDA 12.1 packages for PyG, DGL and PyTorch available on conda.

**Comment by jakirkham** [comment]
Think the packages should be compatible with CUDA 12.2. Let's give this another try

**Comment by alexbarghi-nv** [comment]
@jakirkham both the PyG and DGL tests for CUDA 12.2 failed because of the CUDA version incompatibility.

**Comment by alexbarghi-nv** [comment]
Mamba wasn't able to install the right packages because we had a conflict between packages for CUDA 12.2 and CUDA 12.1.  I'm not sure how we can resolve this in conda.

**Comment by bdice** [comment]
@alexbarghi-nv @jakirkham Perhaps this can help (copying from an internal discussion). You can just install CUDA 12.1 from conda-forge to align with PyTorch's CUDA version.

> Try this. PyTorch only has CUDA 12.1 builds on the pytorch channel, so you probably need to match to that. RAPIDS will work with CUDA 12.1, even though the install page only lists 12.0 and 12.2.
> ```
> mamba create -n pytorch-rapids-cuda-12.1 rapids pytorch torchvision torchaudio pytorch-cuda=12.1 cuda-version=12.1 -c rapidsai -c pytorch -c conda-forge -c nvidia
> ```
>
> I verified that both cudf and torch can be imported and used:
> ```
> import cudf
> import torch
>
> device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
> arr = torch.ones(100, device=device)
> print(arr)
>
> print(cudf.Series(arr))
> ```

File: ci/test_python.sh [hunk 2]
```diff
@@ -142,151 +142,126 @@ rapids-logger "pytest cugraph-service (single GPU)"
...
-    # Reactivate the test environment back
-    set +u
-    conda deactivate
-    conda activate test
-    set -u
-  else
-    rapids-logger "skipping cugraph_pyg pytest on ARM64"
-  fi
+if [[ "${RUNNER_ARCH}" != "ARM64" ]]; then
+  rapids-mamba-retry env create --yes -n test_cugraph_dgl rapids pytorch "cuda-version=${CONDA_CUDA_VERSION}" -c rapidsai -c conda-forge -c nvidia
```
Comment (jakirkham) [inline ci/test_python.sh:153]:
```suggestion
  rapids-mamba-retry create --yes -n test_cugraph_dgl rapids pytorch "cuda-version=${CONDA_CUDA_VERSION}" -c rapidsai -c conda-forge -c nvidia
```

File: ci/test_python.sh [hunk 3]
```diff
@@ -142,151 +142,126 @@ rapids-logger "pytest cugraph-service (single GPU)"
...
+  set +u
+  conda deactivate
+  conda activate test
+  set -u
+else
+  rapids-logger "skipping cugraph_dgl pytest on ARM64"
+fi
+
+if [[ "${RUNNER_ARCH}" != "ARM64" ]]; then
+  rapids-mamba-retry env create --yes -n test_cugraph_pyg rapids pytorch "cuda-version=${CONDA_CUDA_VERSION}" -c rapidsai -c conda-forge -c nvidia
```
Comment (jakirkham) [inline ci/test_python.sh:193]:
```suggestion
  rapids-mamba-retry create --yes -n test_cugraph_pyg rapids pytorch "cuda-version=${CONDA_CUDA_VERSION}" -c rapidsai -c conda-forge -c nvidia
```

**Review by jakirkham — COMMENTED** [review]
Thanks Alex & Bradley! 🙏

Think we want to tweak the command here

For context:

* `mamba env create` is used with environment files
* `mamba create` is used with a list of packages

Hence the latter in this case

**Comment by alexbarghi-nv** [comment]
Closing this; I think we need to wait until the repository migration, which will pin the RAPIDS version and resolve that problem.

**Pull request closed without merging**